src/check_datapackage/check.py

```diff
@@ -97,6 +101,78 @@ class for more details, especially about the default values.
     return issues
 
 
+def _check_keys(properties: dict[str, Any], issue_jsonpaths: list[str]) -> list[Issue]:
+    """Check that primary and foreign keys exist."""
+    # Primary keys
+    resources_with_pk = _get_fields_at_jsonpath(
+        "$.resources[?(length(@.schema.primaryKey) > 0 || @.schema.primaryKey == '')]",
+        properties,
+    )
+    resources_with_pk = _only_resources_with_correct_property_at(
+        "schema.primaryKey", resources_with_pk, issue_jsonpaths
```

Comment:
```suggestion
        resources_with_pk, jsonpaths, "schema.primaryKey"
```

Revise ordering so main objects we are working with are first in the position.